services/repository/mongodb/service.go

```diff
@@ -0,0 +1,204 @@
+	commonErrors "github.com/micro-business/go-core/system/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type mongodbRepositoryService struct {
+	configurationService configuration.ConfigurationContract
+	collectionName       string
+}
+
+// NewMongodbRepositoryService creates new instance of the RepositoryService, setting up all dependencies and returns the instance
+// Returns the new service or error if something goes wrong
+func NewMongodbRepositoryService(
+	configurationService configuration.ConfigurationContract) (repository.RepositoryContract, error) {
+
+	if configurationService == nil {
+		return nil, commonErrors.NewArgumentNilError("configurationService", "configurationService is required")
+	}
+
+	return &mongodbRepositoryService{
+		configurationService: configurationService,
+		collectionName:       "tenants",
+	}, nil
+}
+
+// ReadTenant read an existing tenant
+// context: Optional The reference to the context
+// request: Mandatory. The request to read an existing tenant
+// Returns either the result of reading an exiting tenant or error if something goes wrong.
+func (service *mongodbRepositoryService) ReadTenant(
+	ctx context.Context,
+	request *repository.ReadTenantRequest) (*repository.ReadTenantResponse, error) {
+
+	dbConnectionString, err := service.configurationService.GetDbConnectionString()
+	if err != nil {
+		return nil, commonErrors.NewArgumentNilError("GetDbConnectionString", "Database connection String is required.")
+	}
+
+	clientOptions := options.Client().ApplyURI(dbConnectionString)
+	client, err := mongo.Connect(ctx, clientOptions)
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("Could not connect to mongodb database.", err)
+	}
+
+	tenantDbName, err := service.configurationService.GetTenantDbName()
+	if err != nil {
+		return nil, commonErrors.NewArgumentNilError("GetTenantDbName", "Database name is required.")
+	}
+
+	collection := client.Database(tenantDbName).Collection(service.collectionName)
+	ObjectID, _ := primitive.ObjectIDFromHex(request.TenantID)
+	filter := bson.D{{"_id", ObjectID}}
+	var tenant models.Tenant
+
+	err = collection.FindOne(ctx, filter).Decode(&tenant)
+	if err != nil {
+		return nil, repository.NewTenantNotFoundError(request.TenantID)
+	}
+
+	return &repository.ReadTenantResponse{Tenant: tenant}, nil
+}
+
+// CreateTenant creates a new tenant.
+// context: Optional The reference to the context
+// request: Mandatory. The request to create a new tenant
+// Returns either the result of creating new tenant or error if something goes wrong.
+func (service *mongodbRepositoryService) CreateTenant(
+	ctx context.Context,
+	request *repository.CreateTenantRequest) (*repository.CreateTenantResponse, error) {
+
+	dbConnectionString, err := service.configurationService.GetDbConnectionString()
```

Comment:
Read the connection string once on top of the file and just add the connection string to the service context, no need to read it each time by accessing the config service